include/alpaka/kernel/Traits.hpp

```diff
@@ -237,6 +250,13 @@ namespace alpaka
     {
         // check for void return type
         detail::CheckFnReturnType<TAcc>{}(kernelFnObj, args...);
+#if defined(__cpp_fold_expressions) && defined(__cpp_lib_type_trait_variable_templates)
```

Comment:
Is there a benefit to having both variants? Shouldn't we use the version which works for all compilers?

Reply:
We should just keep the fold expression and move on. But some users still require C++14, so I provide the backport. That backport should be deleted as soon as we can have fold expressions. The reason I keep both versions is that if I just keep the C++14 version, someone else needs to figure out the fold expression and do the refactoring. And I started with the fold expression, so the code was already written.